### ISSUE TITLE
Commenting CSS

### ISSUE THREAD

**Issue by michaelspellacy** [tmpworldwide/tmpworldwide.github.io#28]
@dmechlin Asked me what my thoughts were in having UI's add their name/comment to the Sass/CSS, especially the lead on a project, so that we can easily reach out to each other if there are any issue on any given site. I do it out of habit for that reason and because an artist should ALWAYS sign their work. :-) Thoughts? I always found it convenient.

**Comment by michaelspellacy** [comment]
That or humans.txt. :-)

http://humanstxt.org/

**Comment by Brockenstein** [comment]
I think it is only useful if everyone tracks exactly what they did to a site after the original author did the site in this file. Because these days, we can easily see who made the original site in in version history. So I personally don't see the point if its only tracking initial developer. But whatever the team wants to do is fine with me.

I try to make most my comments as // so they don't appear in the production file. I believe though /*! will keep your comment from minifying.

**Comment by stephendelancey** [comment]
In the compressed css file, all comments are taken out. In scss /\* comments are just multi line comments. They will only come out in the compiled css file if set to in the compiler. 

TB compiler is set to extract that unless it changed in the past 4 months or so.

**Comment by dmechlin** [comment]
I want everyone that goes to the site to see the author. Including those who have access to version history and those who do not. I want traffic and management to easily go to any site that we built and quickly identify the person who built it. If we need to know who made a change and have to dig, we can go into version history and dig around. This will really help us when assigning projects and various reporting and researching task. Maybe even a link to the original build request JIRA ticket.

Brock - can we test your non minifying comment solution?

Example

/*!
Title:         Panera Bread - Talentbrew
Author:        TMP Worldwide - Louisville
               Brock Barnett
               brock.barnett@tmp.commailto:brock.barnett@tmp.com
Creation:      2016-09-08
*/

**Comment by michaelspellacy** [comment]
@stephendelancey Non-minifying comments work. Always have. Built into Sass. I'm paranoid about spam, so I always use [at] in place of @, but whatever you guys want. :-) I use this on tmp to show our font licensing copyright, etc.  

/*!
      Title: Panera Bread - Talentbrew
  Lead: Brock Barnett (brock.barnett[at]tmp.com)
  Contributors: Rebecca Hammer (rebecca.hammer[at]tmp.com), Michael Spellacy (michael.spellac[at]tmp.com)
Creation: 2016-09-08
*/

I don't think location is necessary, We want this to be as easy for devs to drop in or update as is possible. 

Is creation the launch date?

**Comment by Brockenstein** [comment]
Creation should be when the site is either started by the original dev or at latest sent to QA. If its launch date that could be a year from actual creation.

Also is there any real reason to have the emails? Name alone should work fine as it is intended really for internal use it sounds like.

**Comment by lenawan** [comment]
I usually add the Author information on the TB1.0 sites that includes my
email address. But I got a lot of junk emails. Now I'm still receiving
emails from a unknown person.
I think the author's information and the changing history is important for
the maintenance and troubleshooting. But, no email address. May use the
user name instead of the developers name.

**Comment by michaelspellacy** [comment]
Yeah, I agree. Email can be optional. Okay on date.

**Comment by michaelspellacy** [comment]
@lenawan  I don't get a ton of spam, so perhaps [at] is a decent deterrent. If you do include email, just use that replacement, instead. It might help.

**Comment by lenawan** [comment]
That's a good idea!

**Comment by stephendelancey** [comment]
It sounds like adding a comment block to scss won't help solve Dave's problem. Scss is less visible then version history is. 

Spell, just for clarity sake this is where my brain is for comments:
// and /\* are scss single and multi line comments that do not appear in compiled css.

@\* comments are razor/c# comments there for only server side. 

<!-- are html comments which could be compiled out depending if html has any formore of compression. I do not know it in the system or not.

But maybe in the site notes there could be a running list of developing who worked on the site.

**Comment by Brockenstein** [comment]
@stephendelancey  try

```
/*!
This multiline comment
place it at the top of your scss
It should not minify
due to the exclamation mark
*/
```

**Comment by lenawan** [comment]
I added this comments to my current project:
/*
Author: @lenaW
Ticket: TCDQ-15552
Completed on: 09-13-2016
DPM: angelaG & kwasaM
QA:
Changed on:
=============================================*/

**Comment by michaelspellacy** [comment]
Including your JIRA handle not something I considered. Good idea. :-)

**Comment by lenawan** [comment]
We can track everything with ticket number. Then we can shorten the
comments.Maybe something like this:

(1)
/_TCDQ-15552_/

(2)
/*
Author: @lenaW
Ticket: TCDQ-15552
Completed on: 09-13-2016
===============================*/

**Comment by lenawan** [comment]
but the changing history should be appended after that

**Comment by stephendelancey** [comment]
http://stackoverflow.com/questions/16844408/which-output-styles-remove-multiline-line-comments

Answer to scss compiling 
You guys are using compressed so the only way to keep multi line comments is to make them loud with !
So it would be /\* ! I am a loud comment*/

**Comment by michaelspellacy** [comment]
@lenawan That is fine, you just have to add the ! to the comment, as indicated above in various comments. This is so it is not stripped and can be easily accessed by all.

**Comment by michaelspellacy** [comment]
Again, in making this simple, I think all we need here is:

/*!
Title: Panera Bread - Talentbrew
Lead: Brock Barnett (brock.barnett[at]tmp.com, @bbarnett) 
Contributors: 
Rebecca Hammer (rebecca.hammer[at]tmp.com)
Michael Spellacy (michael.spellac[at]tmp.com)
Ticket: TB-123
*/

Email addresses are optional. Since Creation is more a date range, do we even need to include? Ticket enough to provide that detail, I think. This is just quick overview of info.

**Comment by dmechlin** [comment]
Yes, perfect!

Dave

812-989-9993

On Sep 13, 2016, at 12:15 PM, michaelspellacy <notifications@github.com<mailto:notifications@github.com>> wrote:

Again, in making this simple, I think all we need here is:

/*!
Title: Panera Bread - Talentbrew
Lead: Brock Barnett (brock.barnett[at]tmp.comhttp://tmp.com, @bbarnetthttps://github.com/bbarnett)
Contributors:
Rebecca Hammer (rebecca.hammer[at]tmp.comhttp://tmp.com)
Michael Spellacy (michael.spellac[at]tmp.comhttp://tmp.com)
Ticket: TB-123
*/

Email addresses are optional. Since Creation is more a date range, do we even need to include? Ticket enough to provide that detail, I think. This is just quick overview of info.

##

**Comment by lenawan** [comment]
oh yes.
I added.
https://tbcdn.talentbrew.com/company/1769-preview-site/css/3210-Full.css
thank for reminding me!

**Comment by bbarnett** [comment]
you guys should probably take me off this email thread. I'm not the 
"barnett" you're looking for.

**Comment by michaelspellacy** [comment]
@bbarnett Ha! ha! Sorry about that!

**Comment by Brockenstein** [comment]
So I think I am going to go with this for my site. As it seems easiest on my eyes to read. I am dropping the email and only doing the jira tag. 

```
/*!

          Title: Panera Bread - Talentbrew
         Author: TMP Worldwide - Louisville
 Lead Developer: Brock Barnett (@bbarnett) 
   Contributors: Rebecca Hammer (@rhammer)
                 Michael Spellacy (@mspellacy)
Creation Ticket: TCDQ-15843
  Creation Date: 2016-mm-dd

*/ 
```

I know its already been stated that this is only for creation but we keep running into the problem of trying to figure out who last edited what and why those changes haven't been pushed yet. So maybe we should look back into comments from the Product Support Devs. I am talking about any edits that are made after the original dev makes the site because some sites have tons of changes before launch.

```
/*!
          Title: Panera Bread - Talentbrew
         Author: TMP Worldwide - Louisville
 Lead Developer: Brock Barnett (@bbarnett) 
   Contributors: Rebecca Hammer (@rhammer)
                 Michael Spellacy (@mspellacy)
Creation Ticket: TCDQ-15843
  Creation Date: 2016-mm-dd

Post-Creation Changes

         Ticket: TCDQ-15844
      Developer: Adam Taylor (@actaylor)
        Manager: Pss or DPM name
    Change date: 2016-mm-dd
   Dev comments: New Banner Images

*/
```

**Comment by michaelspellacy** [comment]
I'm down with this, but I don't think Author is required as it is implied. For contact method, feel free to add JIRA handle, but use whatever the best way to contact you may be.

Here is the thing, though. When it comes to changes to a site, we do more than just edit CSS, right? It would be _really nice_ if we could come up with a process for implementing a changelog. Something like jobs.sitename.com/changelog

**Comment by michaelspellacy** [comment]
Format similar to this: http://keepachangelog.com/en/0.3.0/

**Comment by michaelspellacy** [comment]
We would need to create it, per site, naturally, but would really need to get people into the habit of maintaining it. Hard to do, but perhaps no more difficult than getting folks to update the comments in the CSS. I mean, as good developer habits go, it behooves you to keep a log no matter where it may be.
